pathiomock.go

```diff
@@ -0,0 +1,59 @@
+package pathio
```

Comment:
What are your thoughts on putting this in a sub-package? Seems like it'd only be used for tests